datamonster_api/lib/datamonster.py

```diff
@@ -64,15 +68,23 @@ def get_company_by_ticker(self, ticker):
     def get_company_by_id(self, company_id):
         """Get a single company by id
 
-        :param company_id: company_id to search for
+        :param company_id: (str) company_id to search for, str form of pk, e.g. '718'
```

Comment:
Yeah, why are we doing this?

Reply:
Both `Company` and `Datasource` subclass `BaseClass`, which defines `self.id` and sets it in `__init__`. The `id` attribute is a string.
There's also a `Datamonster.get_company_by_pk(pk: int)` method, has been there since I first saw this code.

Reply:
I'm confused `BaseClass` doesn't even define an `__init__`.

Reply:
Ah, true that - wishful thinking on my part: its subclasses all (both) define identical `__init__`s. 

In any case, when one calls `company = dm.get_company_by_id(co_id)`, the arg `co_id` can be a `str` or an `int` [1], *but `company.id` will be a `str`*.

[*] By a happy accident, whether `str` or `int` doesn't matter, as `co_id` is just passed to a simple untyped placeholder `{}` in a format string, part of the REST endpoint path to the company's details.